translator/translate/otel/exporter/awsemf/translator.go

```diff
@@ -142,14 +140,27 @@ func (t *translator) Translate(c *confmap.Conf) (component.Config, error) {
 
 func getAppSignalsConfig() string {
 	ctx := context.CurrentContext()
-	kubernetesMode := ctx.KubernetesMode()
 
-	if kubernetesMode == config.ModeEKS {
+	mode := ctx.KubernetesMode()
+	if mode == "" {
+		mode = ctx.Mode()
+	}
+	if mode == config.ModeEC2 {
+		if ecsutil.GetECSUtilSingleton().IsECS() {
+			mode = config.ModeECS
+		}
+	}
+
+	switch mode {
+	case config.ModeEKS:
 		return appSignalsConfigEks
-	} else if kubernetesMode == config.ModeK8sEC2 || kubernetesMode == config.ModeK8sOnPrem {
+	case config.ModeK8sEC2, config.ModeK8sOnPrem:
 		return appSignalsConfigK8s
+	case config.ModeEC2, config.ModeECS:
+		return appSignalsConfigGeneric
```

Comment:
Should ModeEC2 have an EC2 config? Don't we have an EC2 resolver?

Reply:
The config template is actually unified regardless of the platform. The resolver is mainly for different attributes population.